packages/app/src/debug/DebugFailedTest.cy.tsx

```diff
@@ -1,37 +1,195 @@
+      instance: {
+        id: '123',
+        groupId: '123',
+        status: 'FAILED',
+        hasScreenshots: false,
+        hasStdout: false,
+        hasVideo: false,
+      },
     }
 
     cy.mount(() => (
-      <DebugFailedTest failedTestResult={testResult} />
+      <div data-cy="test-group">
+        <DebugFailedTest failedTestsResult={[testResult]} groups={[group1]} expandable={false}/>
+      </div>
     ))
 
-    cy.findByTestId('test-row').children().should('have.length', 2)
+    cy.findByTestId('test-row').children().should('have.length', 6)
+    cy.findByTestId('failed-icon').should('be.visible')
+    rowElementTesting(testResult)
 
-    testResult.titleParts.forEach((title, index) => {
-      cy.findByTestId(`titleParts-${index}`).should('have.text', `${title}`)
-    })
+    cy.findByTestId('test-group').realHover()
+    cy.findByTestId('debug-artifacts').should('be.visible').children().should('have.length', 3)
+
+    cy.percySnapshot()
   })
 
-  it('contains multiple titleParts segments', () => {
-    const multipleTitleParts = {
+  it('contains multiple titleParts segments', { viewportWidth: 1200 }, () => {
+    const multipleTitleParts: TestResults = {
       id: '676df87878',
       titleParts: ['Login', 'Describe', 'it', 'context', 'Should redirect unauthenticated user to signin page'],
+      instance: {
+        id: '456',
+        groupId: '456',
+        status: 'FAILED',
+        hasScreenshots: false,
+        hasStdout: false,
+        hasVideo: false,
+      },
     }
 
     cy.mount(() => (
-      <DebugFailedTest failedTestResult={multipleTitleParts} />
+      <DebugFailedTest failedTestsResult={[multipleTitleParts]} groups={[group1]} expandable={false}/>
     ))
 
-    cy.findByTestId('test-row').children().should('have.length', 5).should('be.visible')
-    multipleTitleParts.titleParts.forEach((title, index) => {
-      cy.findByTestId(`titleParts-${index}`).should('have.text', `${title}`)
-    })
+    rowElementTesting(multipleTitleParts)
+
+    cy.percySnapshot()
+  })
+
+  it('tests multiple groups', { viewportWidth: 1200 }, () => {
+    const testResults: TestResults[] = [
+      {
+        id: '676df87878',
+        titleParts: ['Login', 'Describe', 'it', 'context', 'Should redirect unauthenticated user to signin page'],
+        instance: {
+          id: '456',
+          groupId: '456',
+          status: 'FAILED',
+          hasScreenshots: false,
+          hasStdout: false,
+          hasVideo: false,
+        },
+      },
+      {
+        id: '676df87878',
+        titleParts: ['Login', 'Should redirect unauthenticated user to signin page'],
+        instance: {
+          id: '123',
+          groupId: '123',
+          status: 'FAILED',
+          hasScreenshots: false,
+          hasStdout: false,
+          hasVideo: false,
+        },
+      },
+    ]
+
+    cy.mount(() => (
+      <DebugFailedTest failedTestsResult={testResults} groups={[group1, group2]} expandable={true}/>
+    ))
+
```

Comment:
Add an assertion that the artifact links are not present on the row with the test title since they are shown per group.